### PR TITLE
Add shub capability to shield iks01a2 

### PR DESCRIPTION
**Pull request by avisconti**
Add sensorhub capability to shield iks01a2.
This change to boards/shields and samples/shields was used to test #28591 Pull Request.

### CODE REVIEW

File: samples/shields/x_nucleo_iks01a2/sensorhub/src/main.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2018 STMicroelectronics
```
Comment (erwango) [inline samples/shields/x_nucleo_iks01a2/sensorhub/src/main.c:2]:
```suggestion
 * Copyright (c) 2020 STMicroelectronics
```

**Review by erwango — CHANGES_REQUESTED** [review]
Few nitpicks

File: boards/shields/x_nucleo_iks01a2/x_nucleo_iks01a2_shub.overlay [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+ *
+ * Copyright (c) 2018 Linaro Limited
```
[the remark this answers is not in the document]
Comment (avisconti) [reply boards/shields/x_nucleo_iks01a2/x_nucleo_iks01a2_shub.overlay]: ack

File: samples/shields/x_nucleo_iks01a2/sensorhub/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+
+# This sample is specific to x_nucleo_iks01a2 shield. Enforce -DSHIELD option
+set(SHIELD x_nucleo_iks01a2_shub)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(x_nucleo_iks01a2)
```
[the remark this answers is not in the document]
Comment (avisconti) [reply samples/shields/x_nucleo_iks01a2/sensorhub/CMakeLists.txt]: thx

**Review by erwango — CHANGES_REQUESTED** [review]
I think you need some tweaks in shub sample yaml file

File: samples/shields/x_nucleo_iks01a2/standard/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+sample:
+  name: X-NUCLEO-IKS01A2 sensor shield
+tests:
+  sample.shields..x_nucleo_iks01a2:
```
[the remark this answers is not in the document]
Comment (avisconti) [reply samples/shields/x_nucleo_iks01a2/standard/sample.yaml:4]: This one looks more a fix of a previous existing bug. Should I include it in the same commit?

[the remark this answers is not in the document]
Comment (avisconti) [reply samples/shields/x_nucleo_iks01a2/standard/sample.yaml:4]: OK, I'll include it in the same commit.

File: boards/shields/x_nucleo_iks01a2/x_nucleo_iks01a2_shub.overlay (hunk 2 above)
[the remark this answers is not in the document]
Comment (avisconti) [reply boards/shields/x_nucleo_iks01a2/x_nucleo_iks01a2_shub.overlay]: fixed, thanks!

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**